src/StatsdClient/Serializer/SerializerHelper.cs

```diff
@@ -0,0 +1,81 @@
+using System.Collections.Concurrent;
+using System.Text;
+
+namespace StatsdClient
+{
+    internal class SerializerHelper
+    {
+        private static readonly string[] EmptyArray = new string[0];
+        private readonly ConcurrentQueue<SerializedMetric> _pool = new ConcurrentQueue<SerializedMetric>();
```

Comment:
There is no upper bound on this queue, right? It should eventually reach a state of equilibrium, but that state is dependent on load. I'm wondering if this is perhaps too optimistic an approach?

Reply:
Indeed, if the processing of the metric is not fast enough, it can lead to out of memory. I will fix it.

Reply:
I am working on creating a generic pool to limit the number of allocated objects. I am doing it in a separate PR as it introduce several changes.

Reply:
https://github.com/DataDog/dogstatsd-csharp-client/pull/132